controllers/scopeinstance_controller.go

```diff
@@ -72,43 +84,42 @@ func (r *ScopeInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	st := &operatorsv1.ScopeTemplate{}
 	if err := r.Client.Get(ctx, client.ObjectKey{Name: in.Spec.ScopeTemplateName}, st); err != nil {
 		if !k8sapierrors.IsNotFound(err) {
```

Comment:
I think this will not delete role bindings if ScopeTemplate is not present. It should be like below code.

```
                  if !k8sapierrors.IsNotFound(err) {
			

		listOption := client.MatchingLabels{
			scopeInstanceUIDKey: string(in.GetUID()),
		}

		if err := r.deleteBindings(ctx, listOption); err != nil {
			log.Log.Info("Error in deleting Role Bindings", "error", err)
			return ctrl.Result{}, err
		}
		}
                return ctrl.Result{}, err
}

```

Reply:
I don't think that's correct. We should delete the roleBindings and clusterRoleBindings if we hit a `not found error`, but we shouldn't delete them because of unknown errors, such as an interruption to the API server.

Reply:
The existing if statement simply exits early if an error exists and IS NOT an ISNotFound error.

Reply:
Ah! git it sorry my bad I saw it another way around.